common/ledger/blkstorage/blockfile_mgr.go

```diff
@@ -471,9 +527,12 @@ func (mgr *blockfileMgr) retrieveBlockByNumber(blockNum uint64) (*common.Block,
 
 func (mgr *blockfileMgr) retrieveBlockByTxID(txID string) (*common.Block, error) {
 	logger.Debugf("retrieveBlockByTxID() - txID = [%s]", txID)
-
 	loc, err := mgr.index.getBlockLocByTxID(txID)
-
+	if err == errNilValue {
+		return nil, errors.Errorf(
+			"details for the TXID [%s] not available. Ledger bootstrapped from a snapshot. First available block = [%d]",
+			txID, mgr.firstPossibleBlockNumberInBlockFiles())
```

Comment:
an exported error type? the caller shouldn't panic or stop a chain. The same goes for 4 below errors.

Reply:
same as above